velox/core/PlanNode.h

```diff
@@ -1731,16 +1731,26 @@ class OrderByNode : public PlanNode {
       const std::vector<SortOrder>& sortingOrders,
       bool isPartial,
       const PlanNodePtr& source)
-      : PlanNode(id),
-        sortingKeys_(sortingKeys),
-        sortingOrders_(sortingOrders),
-        isPartial_(isPartial),
-        sources_{source} {
+      : PlanNode(id), isPartial_(isPartial), sources_{source} {
     VELOX_USER_CHECK(!sortingKeys.empty(), "OrderBy must specify sorting keys");
     VELOX_USER_CHECK_EQ(
         sortingKeys.size(),
         sortingOrders.size(),
         "Number of sorting keys and sorting orders in OrderBy must be the same");
+    std::vector<FieldAccessTypedExprPtr> deduplicatedSortingKeys;
+    std::vector<SortOrder> deduplicatedSortingOrders;
+    std::unordered_set<std::string> sortingKeySet;
+    // Eliminate duplicated sorting keys.
```

Comment:
Instead of eliminating duplicate keys, let's just reject duplicates and document that duplicate keys are not allowed.

Reply:
addressed